main.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+require 'json'
+require 'sinatra'
+require 'securerandom'
+
+before do
+  @app_title = 'メモアプリ'
+end
+
+helpers do
+  def h(text)
+    Rack::Utils.escape_html(text)
+  end
+end
+
+class MemoDB
+  DB_FILE_NAME = 'db.json'
+
+  def self.read_memos
+    load_memos
+  end
+
+  def self.create_memo(title, content)
+    memos = load_memos
+    id = SecureRandom.uuid
+
+    memos[id] = { 'title' => title, 'content' => content }
+    save_memo(memos)
+  end
+
+  def self.read_memo(id)
+    memos = load_memos
+    memos[id.to_sym]
+  end
+
+  def self.update_memo_details(params)
+    memos = load_memos
+
+    memos[params['id'].to_sym] = params.slice('title', 'content')
+    save_memo(memos)
+  end
+
+  def self.delete_memo(params)
+    memos = load_memos
+
+    memos.delete(params['id'].to_sym)
+    save_memo(memos)
+  end
+
+  def self.save_memo(memos)
+    File.open(DB_FILE_NAME, 'w') do |file|
+      converted_memos = JSON.generate(memos)
+      file.write(converted_memos)
+    end
+  end
+
+  def self.load_memos
+    file = File.open(DB_FILE_NAME, 'r')
+    JSON.parse(file.read, symbolize_names: true)
+  end
+end
+
+get '/' do
+  redirect '/memos'
+end
+
+get '/memos' do
+  @memos = MemoDB.read_memos
+  erb :index
+end
+
+get '/memos/create' do
+  erb :create
+end
+
+post '/memos/new' do
+  MemoDB.create_memo(@params['title'], @params['content'])
```

Comment:
updateと同様にハッシュを渡してしまった方がいいかも。
将来的に項目が増えると、引数もどんどん増えていってしまうので。

```suggestion
  MemoDB.create_memo(@params)
```